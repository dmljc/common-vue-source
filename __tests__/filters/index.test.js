/**
 * @jest-environment jsdom
 */
import Vue from "vue";
import Compose from '@/main';
import TraySizeFilter from './templates/tray.size';
import ContainerTypeFilter from './templates/container.type';
import SelectionNamesFilter from './templates/selection.names';
import SecondStrFilter from './templates/second.str';
import CascaderNamesFilter from './templates/cascader.names';

beforeAll(() => {
    Vue.use(Compose);
});
describe('filter.js lsTraySizeStr', () => {
    let cmp, vm;
    beforeAll(() => {
        cmp = Vue.extend(TraySizeFilter);
        vm = new cmp({
            data: {}
        }).$mount();
    });
    it('equals lsTraySizeStr to snapshot', () => {
        expect(vm.$el).toMatchSnapshot();
    });
});

describe('filter.js containerTypeList', () => {
    let cmp, vm;
    beforeAll(() => {
        cmp = Vue.extend(ContainerTypeFilter);
        vm = new cmp({
            data: {}
        }).$mount();
    });
    it('equals containerTypeFilter to snapshot', () => {
        expect(vm.$el).toMatchSnapshot();
    });
});

describe('filter.js selectionNames', () => {
    let cmp, vm;
    beforeAll(() => {
        cmp = Vue.extend(SelectionNamesFilter);
        vm = new cmp({
            data: {}
        }).$mount();
    });
    it('equals selectionNamesFilter to snapshot', () => {
        expect(vm.$el).toMatchSnapshot();
    });
});

describe('filter.js SecondStrFilter', () => {
    let cmp, vm;
    beforeAll(() => {
        cmp = Vue.extend(SecondStrFilter);
        vm = new cmp({
            data: {}
        }).$mount();
    });
    it('equals SecondStrFilter to snapshot', () => {
        expect(vm.$el).toMatchSnapshot();
    });
});

describe('filter.js CascaderNamesFilter', () => {
    let cmp, vm;
    beforeAll(() => {
        cmp = Vue.extend(CascaderNamesFilter);
        vm = new cmp({
            data: {}
        }).$mount();
    });
    it('equals CascaderNamesFilter to snapshot', () => {
        expect(vm.$el).toMatchSnapshot();
    });
});