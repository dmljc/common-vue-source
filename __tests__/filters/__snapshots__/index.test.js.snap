// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter.js CascaderNamesFilter equals CascaderNamesFilter to snapshot 1`] = `
<div>
  <span>
    中国/杭州
  </span>
   
  <span>
    中国/杭州
  </span>
   
  <span>
    中国/杭州
  </span>
   
  <span>
    
  </span>
   
  <span>
    
  </span>
   
  <span>
    法国/巴黎
  </span>
   
  <span>
    --
  </span>
   
  <span>
    --
  </span>
   
  <span>
    --
  </span>
   
  <span>
    
  </span>
   
  <span>
    中国
  </span>
</div>
`;

exports[`filter.js SecondStrFilter equals SecondStrFilter to snapshot 1`] = `
<div>
  <span>
    00:00:00
  </span>
   
  <span>
    00:00:-1
  </span>
   
  <span>
    00:01:00
  </span>
   
  <span>
    01:00:00
  </span>
   
  <span>
    23:59:59
  </span>
   
  <span>
    00:00:00
  </span>
</div>
`;

exports[`filter.js containerTypeList equals containerTypeFilter to snapshot 1`] = `
<div>
  <span>
    
  </span>
   
  <span>
    
  </span>
   
  <span>
    
  </span>
   
  <span>
    1*20GP, 2*40GP, 3*20RF, 4*40RF, 5*40HC, 6*45HC
  </span>
   
  <span>
    1*20GP, 2*40GP, 3*20RF
  </span>
   
  <span>
    1*20GP, 0*40GP, *20RF
  </span>
</div>
`;

exports[`filter.js lsTraySizeStr equals lsTraySizeStr to snapshot 1`] = `
<div>
  <span>
    --
  </span>
   
  <span>
    --
  </span>
   
  <span>
    --
  </span>
   
  <span>
    0 cm * 0 cm * 0 cm * 0 件
  </span>
   
  <span>
    0 cm * 0 cm * 0 cm * 0 件, 99 cm * 99 cm * 99 cm * 99 件
  </span>
</div>
`;

exports[`filter.js selectionNames equals selectionNamesFilter to snapshot 1`] = `
<div>
  <span>
    --
  </span>
   
  <span>
    中国, 美国
  </span>
   
  <span>
    --
  </span>
   
  <span>
    中国, 美国
  </span>
   
  <span>
    , 
  </span>
   
  <span>
    , 
  </span>
   
  <span>
    , 
  </span>
   
  <span>
    中国, , , 美国
  </span>
</div>
`;
